core/src/avm2/multiname.rs

```diff
@@ -163,33 +160,57 @@ impl<'gc> Multiname<'gc> {
         let ns_set = ns_set?;
 
         if ns_set.len() == 1 {
-            Ok(NamespaceSet::single(
-                translation_unit.pool_namespace(ns_set[0], context)?,
-            ))
+            let namespace = translation_unit.pool_namespace(activation, ns_set[0])?;
+
+            if namespace.is_any() {
+                return Err(make_error_1080(activation));
+            }
+
+            Ok(NamespaceSet::single(namespace))
         } else {
             let mut result = Vec::with_capacity(ns_set.len());
             for ns in ns_set {
-                result.push(translation_unit.pool_namespace(*ns, context)?)
+                let namespace = translation_unit.pool_namespace(activation, *ns)?;
+
+                // Namespace sets must not have Any namespaces in them
+                if namespace.is_any() {
+                    return Err(make_error_1080(activation));
+                }
+
+                result.push(namespace)
             }
-            Ok(NamespaceSet::multiple(result, context.gc_context))
+
+            Ok(NamespaceSet::multiple(result, activation.gc()))
         }
     }
 
     pub fn from_abc_index(
+        activation: &mut Activation<'_, 'gc>,
         translation_unit: TranslationUnit<'gc>,
         multiname_index: Index<AbcMultiname>,
-        context: &mut UpdateContext<'gc>,
     ) -> Result<Self, Error<'gc>> {
-        let mc = context.gc_context;
+        let mc = activation.gc();
+
+        if multiname_index.0 == 0 {
+            return Err(make_error_1032(activation, 0));
+        }
+
         let abc = translation_unit.abc();
-        let abc_multiname = Self::resolve_multiname_index(&abc, multiname_index)?;
+
+        let abc_multiname = abc
+            .constant_pool
+            .multinames
+            .get(multiname_index.0 as usize - 1)
+            .ok_or_else(|| format!("Unknown multiname constant {}", multiname_index.0))?;
```

Comment:
there's no corresponding error in FP?

Reply:
There is, I'm going to fix all the remaining constant pool errors in a follow-up PR (should just be replaced with `make_error_1032`, like in `verify.rs`).